src/Credentials/CredentialProvider.php

```diff
@@ -692,6 +702,52 @@ private static function loadDefaultProfiles() {
         return $profiles;
     }
 
+    public static function getAssumeRoleCredentials(
+        $sourceProfileName = '',
+        $profileName = '',
+        $filename = '',
+        $config = []
+    )
+    {
+        $data = self::loadProfiles($filename);
+        $credentialSource = !empty($data[$profileName]['credential_source']) ? $data[$profileName]['credential_source'] : null;
+        $credentialsPromise = null;
```

Comment:
Should this statement be inside the `if` block, since it's not used outside of it?

Reply:
Removed the if block since it was no longer necessary